app/src/main/java/net/gsantner/markor/format/TextFormat.java

```diff
@@ -90,8 +90,9 @@ public static TextFormat getFormat(int formatId, Activity activity, Document doc
             default:
             case FORMAT_MARKDOWN: {
                 format.setConverter(CONVERTER_MARKDOWN);
-                format.setHighlighter(new MarkdownHighlighter(hlEditor, document));
-                format.setTextActions(new MarkdownTextActions(activity, document));
+                MarkdownTextActions mta = new MarkdownTextActions(activity, document);
+                format.setTextActions(mta);
+                format.setHighlighter(new MarkdownHighlighter(hlEditor, document, mta));
```

Comment:
See comment below. I feel this is not very clean. Not sure if 'update as you type' is needed at all.